packages/core/src/shareback/cda-uploader.ts

```diff
@@ -0,0 +1,71 @@
+import { Organization } from "@medplum/fhirtypes";
+import { createUploadMetadataFilePath } from "../domain/document/upload";
+import { MetriportError } from "../util/error/metriport-error";
+import { createAndUploadCDAMetadataFile } from "./create-and-upload-extrinsic-object";
+import { createUploadFilePath } from "../domain/document/upload";
+import { uuidv7 } from "../util/uuid-v7";
+import { S3Utils } from "../external/aws/s3";
+import { out } from "../util/log";
+
+const MAXIMUM_FILE_SIZE = 50_000_000; // 50 MB
+const { log } = out(`Core CDA Uploader`);
+
+export async function cdaDocumentUploaderHandler({
+  cxId,
+  patientId,
+  cdaBundle,
+  destinationBucket,
+  region,
+  organization,
+}: {
+  cxId: string;
+  patientId: string;
+  cdaBundle: string;
+  destinationBucket: string;
+  region: string;
+  organization: Organization;
+}): Promise<void | { message: string; size: number }> {
+  const s3Utils = new S3Utils(region);
+
+  const docId = uuidv7();
+  const metadataFileName = createUploadMetadataFilePath(cxId, patientId, docId);
+  const destinationKey = createUploadFilePath(cxId, patientId, docId);
+
+  // Make a copy of the file to the general medical documents bucket
+  try {
+    await s3Utils.uploadFile(destinationBucket, `${destinationKey}.xml`, Buffer.from(cdaBundle));
```

Comment:
This seems dangerous. We store `.xml` files on the same folder. If we call this function by accident on a file we download from a HIE we will overwrite it.

Also, let's keep S3 file naming consistent, create a function on `packages/core/src/domain/filename.ts` or use like we do on `packages/core/src/domain/document/filename.ts` to create a function for this situation here.

Also, don't we want to store a history of generated files? Like:
- cx pushes some consolidated data
- we generate a CDA
- a few months later they push more consolidated data
- we generate a new CDA (don't overwrite the previous one so we keep a papertrail/traceability)

Reply:
I'm not following... 
The `destinationKey` is already generated using a shared function that also puts the file into `/uploads`: 
```const destinationKey = createUploadFilePath(cxId, patientId, docId);```

Reply:
and the `docId` is generated every time with `uuidv7()`, making each upload unique - so no files are being overwritten here

Reply:
Yeah, I completely missed the context here. I was thinking we were reusing some existing `key`, which is not the case.

[PEBCAK](https://en.wiktionary.org/wiki/PEBCAK)